### ISSUE TITLE
Swagger / OpenAPI documentation / playground

### ISSUE THREAD

**Issue by netandreus** [jmorganca/ollama#716]
Do you plan to generate API documentation with ```github.com/swaggo/swag```?
It will be very usefull.

**Comment by technovangelist** [comment]
Take a look at this from postman that I think addresses your issue - https://www.postman.com/postman-student-programs/workspace/ollama-api/collection/21521806-f48dc31a-a9f1-4dad-9082-fd07f5cd2fda. I will go ahead and close it now. If you think there is anything we left out, reopen and we can address. Thanks for being part of this great community.

**Comment by tvanier** [comment]
For those reading this, there is a (curated?) OpenAPI spec available here (thanks @walsha2, @mthongvanh and @davidmigloz)
https://github.com/davidmigloz/langchain_dart/blob/main/packages/ollama_dart/oas/ollama-curated.yaml

@technovangelist It would be great to keep an "official" API spec in the ollama repo?

**Comment by bidek** [comment]
@tvanier, thanks for the link.

@technovangelist, I checked this PR: https://github.com/jmorganca/ollama/pull/714 and read through the issues. I'd like to add my two cents.

From my perspective, it would be best if Ollama itself could provide an endpoint with OpenAPI definition. This way, it will always be in sync with the code base, and generating client code would be easier.

I'm a developer with a Java background, and in our world, it's the de facto standard for services to provide API spec definitions. To be fair, in Java, we use reflection APIs to generate this spec for us. But maybe there is a similar solution in the GIN framework.

In conclusion, my main concern is making client code generation easier and staying in sync with the newest changes in the API.
